datafusion/sql/tests/sql_integration.rs

```diff
@@ -2671,13 +2673,62 @@ fn logical_plan_with_dialect_and_options(
     dialect: &dyn Dialect,
     options: ParserOptions,
 ) -> Result<LogicalPlan> {
-    let context = MockContextProvider::default();
+    let context = MockContextProvider::default().with_udf(make_udf(
+        "nullif",
+        vec![DataType::Int32, DataType::Int32],
+        DataType::Int32,
+    ));
+
     let planner = SqlToRel::new_with_options(&context, options);
     let result = DFParser::parse_sql_with_dialect(sql, dialect);
     let mut ast = result?;
     planner.statement_to_plan(ast.pop_front().unwrap())
 }
 
+fn make_udf(name: &'static str, args: Vec<DataType>, return_type: DataType) -> ScalarUDF {
```

Comment:
The sql integration test needs function definitions, to resolve function names, but doesn't actually invoke them